ols/app/models/config.py

```diff
@@ -552,6 +553,22 @@ def validate_yaml(self) -> None:
             _dir_check(self.embeddings_model_path, "Embeddings model path")
 
 
+class UserDataCollection(BaseModel):
+    """User data collection configuration."""
+
+    disabled: bool = True
+    storage_location: Optional[DirectoryPath] = None
+
+    @model_validator(mode="after")
```

Comment:
are we mixing the "old" and "new" approach of pydantic model there?

Reply:
Are you talking about models in general or about this model specifically? This one should be just "new".

Reply:
just about this model

Reply:
assuming the answer is "yes", i am fine w/ us starting to introduce pydantic validation in new config logic.  If we think it's better, there's no reason not to start using it where we can, and then go back and refactor the old code later when we don't have more pressing concerns.